src/modules/claims/claims.service.ts

```diff
@@ -1422,6 +1422,9 @@ export class ClaimsService {
       await this._signerService.signMessage(arrayify(proofHash))
     );
   }
+  claimIsExpired(date: number): boolean {
+    return !!date && date < Date.now();
+  }
```

Comment:
I think this function is more useful if it checks a JWT payload, rather than just the date check.
Extracting `Date.now()` give more flexibility to the client and make this function pure.

Another question to think about is if `date` should be a `Date` or a `number`

```suggestion
  jwtIsExpired(payload: {Whatever the correct type is for JWT payload}, date: number): boolean {
    return payload?.exp && date && payload.exp < date;
  }
```

Reply:
so we never will want to use the date from the VC? Only from the JWT payload?

Reply:
Please let me know if you would rather have a date or a number

Reply:
> so we never will want to use the date from a VC? Only from the JWT payload?

No, we might want to check the expiry of a VC as well. My suggestion would be to have 2 methods:
- jwtIsExpired(jwtPayload: jwt.JwtPayload, dateToCheck: Date)
- vcIsExpired(vc: VerifiableCredential, dateToCheck: Date)

> Please let me know if you would rather have a date or a number

I think it is better to use a `Date` type as then there is no ambiguity on what the number should be (e.g. seconds/miliseconds, since when, etc)

Reply:
By the way, I was reading this SO answer that is saying that `exp` from a JWT needs to be multiplied for the check https://stackoverflow.com/a/51293316/6674421

Reply:
But in the use case of switchboard, we won't be passing either a JWT OR a VC to this method - we will just be pulling the date from expirationTimestamp from the claim as it comes to switchboard
`expirationTimestamp: "1658847564357"`

- so can we have another method that just takes in a date number and does a check? And then a method for jwtIsExpired etc...

Reply:
@whitneypurdum 
> we will just be pulling the date from expirationTimestamp from the claim as it comes to switchboard
expirationTimestamp: "1658847564357"

Fair enough, I didn't realize this part. I think we should delete this function then from ICL. The best code is no code at all 😄

Reply:
@jrhender so in the verifyRoleEIP191JWT method, just verify the expiration in the method itself (just a basic number comparison?)

Reply:
Yes, that's what I'm thinking for now. If we see room for refactor, maybe we do that down the line